packages/h5wasm/create_h5_sample.py

```diff
@@ -0,0 +1,215 @@
+
+    # === H5T_BITFIELD ===
+
+    # h5py syntax unknown
+
+    # === H5T_OPAQUE ===
+
+    add_scalar(h5, "byte_string", np.void(b"\x00\x11\x22"))
+    add_scalar(h5, "datetime64", np.void(np.datetime64("2019-09-22T17:38:30")))
+    add_scalar(h5, "datetime64_not-a-time", np.void(np.datetime64("NaT")))
+
+    # === H5T_COMPOUND ===
+
+    add_scalar(
+        h5,
+        "complex64",
+        np.complex64(
+            np.finfo(np.complex64).smallest_normal + np.finfo(np.complex64).max * 1j
+        ),
+    )
+    add_array(
+        h5, "complex64", np.array([[1 + 2j, 3 + 4j], [5 + 6j, 7 + 8j]], np.complex64)
+    ),
+    add_scalar(
+        h5,
+        "complex128",
+        np.complex128(
+            np.finfo(np.complex128).smallest_normal + np.finfo(np.complex128).max * 1j
+        ),
+    )
+    add_scalar(h5, "complex128_BE", 1 + 2j, np.dtype(">c16"))
+    add_array(
+        h5, "complex128", np.array([[1 + 2j, 3 + 4j], [5 + 6j, 7 + 8j]], np.complex128)
+    ),
+    add_scalar(
+        h5,
+        "complex256",
+        np.complex256(
+            np.finfo(np.complex256).smallest_normal + np.finfo(np.complex256).max * 1j
+        ),
+    )
+    add_array(
+        h5, "complex256", np.array([[1 + 2j, 3 + 4j], [5 + 6j, 7 + 8j]], np.complex256)
+    ),
+
+    add_scalar(
+        h5,
+        "compound",
+        (1, 2.0, "foo"),
+        [("bigint", np.int64), ("double", np.float64), ("utf-8", h5py.string_dtype())],
+    )
+    for_ref = add_array(
+        h5,
+        "compound",
+        np.array(
+            [(1, np.nan, "foo"), (2, np.inf, "bar"), (3, np.NZERO, "baz")],
+            [
+                ("bigint", np.int64),
+                ("double", np.float64),
+                ("utf-8", h5py.string_dtype()),
+            ],
+        ),
+    )
+    add_scalar(
+        h5,
+        "compound_nested",
+        ((True, 1 + 2j),),
+        [("nested", [("bool", np.bool_), ("cplx", np.complex64)])],
+    )
+
+    # === H5T_REFERENCE ===
+
+    add_scalar(h5, "reference", for_ref.ref, h5py.special_dtype(ref=h5py.Reference))
+    add_scalar(
+        h5,
+        "reference_region",
+        for_ref.regionref[0:1],
+        h5py.special_dtype(ref=h5py.Reference),
+    )
+
+    # === H5T_ENUM ===
+
+    add_empty(h5, "bool", h5py.enum_dtype({"FALSE": 0, "TRUE": 1}))
+    add_scalar(h5, "bool_false", False)
+    add_scalar(h5, "bool_true", True)
+    add_array(
+        h5, "bool", np.array([[True, False, True, True], [False, False, True, False]])
+    )
+
+    add_scalar(
+        h5,
+        "enum_uint8",
+        1,
+        h5py.enum_dtype({"A": 0, "B": 1}),
+    )
+    add_scalar(
+        h5,
+        "enum_int32",
+        256,
+        h5py.enum_dtype({"A": 256, "B": 257}, np.int32),
+    )
+
+    # === H5T_VLEN ===
+
+    # https://docs.h5py.org/en/stable/special.html#arbitrary-vlen-data
+    scalar_vlen = add_scalar(h5, "vlen_int8", dtype=h5py.vlen_dtype(np.int8))
+    scalar_vlen[()] = [0, 1]
```

Comment:
I'm trying to create a scalar dataset with an `H5T_VLEN` type (itself with an `H5T_INTEGER` base type). Not 100% this is what I'm doing above...